src/Nethermind/Nethermind.Evm/Tracing/ITxTracer.cs

```diff
@@ -16,45 +16,92 @@
-        void MarkAsFailed(Address recipient, long gasSpent, byte[] output, string error);
-
-        void StartOperation(int depth, long gas, Instruction opcode, int pc);
-        void ReportOperationError(EvmExceptionType error);
-        void ReportOperationRemainingGas(long gas);
-        void SetOperationStack(List<string> stackTrace);
-        void ReportStackPush(Span<byte> stackItem);
-        void SetOperationMemory(List<string> memoryTrace);
-        void SetOperationMemorySize(ulong newSize);
-        void ReportMemoryChange(long offset, Span<byte> data);
-        void ReportStorageChange(Span<byte> key, Span<byte> value);
-        void SetOperationStorage(Address address, UInt256 storageIndex, byte[] newValue, byte[] currentValue);
-        
-        void ReportSelfDestruct(Address address, UInt256 balance, Address refundAddress);
-        void ReportAction(long gas, UInt256 value, Address @from, Address to, byte[] input, ExecutionType callType, bool isPrecompileCall = false);
-        void ReportActionEnd(long gas, byte[] output);
-        
-        void ReportActionError(EvmExceptionType evmExceptionType);
-        void ReportActionEnd(long gas, Address deploymentAddress, byte[] deployedCode);
-        void ReportByteCode(byte[] byteCode);
-        void ReportRefundForVmTrace(long refund, long gasAvailable);
-        void ReportRefund(long refund);
+        /// <summary>
+        /// Defines whether MarkAsSuccess or MarkAsFailed will be called
+        /// </summary>
+        bool IsTracingReceipt => false;
+        /// <summary>
+        /// High level calls with information on the target account
+        /// </summary>
+        bool IsTracingActions => false;
+        /// <summary>
+        /// SSTORE and SLOAD level storage operations
+        /// </summary>
+        bool IsTracingOpLevelStorage => false;
+        /// <summary>
+        /// EVM memory access operations
+        /// </summary>
+        bool IsTracingMemory => false;
+        bool IsTracingInstructions => false;
+        /// <summary>
+        /// Code deployment
+        /// </summary>
+        bool IsTracingCode => false;
+        /// <summary>
+        /// EVM stack tracing after each operation
+        /// </summary>
+        bool IsTracingStack => false;
+        /// <summary>
+        /// State changes at commit stage
+        /// </summary>
+        bool IsTracingState => false;
+
+        /// <summary>
+        /// Traces blockhash calls
+        /// </summary>
+        bool IsTracingBlockHash => false;
+
+        void MarkAsSuccess(Address recipient, long gasSpent, byte[] output, LogEntry[] logs, Keccak stateRoot = null) => throw new NotSupportedException();
+
+        void MarkAsFailed(Address recipient, long gasSpent, byte[] output, string error, Keccak stateRoot = null) => throw new NotSupportedException();
+
+        void StartOperation(int depth, long gas, Instruction opcode, int pc) => throw new NotSupportedException();
```

Comment:
throwing NotSupportedException from an interface method implementation looks like some kind of code smell

Reply:
i will remove it